app/helpers/generic_file_helper.rb

```diff
@@ -39,7 +43,22 @@ def download_path(*args)
       gf = GenericFile.find(item.id)
       return gf.doi_url if gf.doi_url.present?
     end
-    return sufia.download_path(*args)
+    if args[1].nil?
```

Comment:
args[1] contains the hash of request attributes - it should only ever be
file=>download - see
https://github.com/ualbertalib/HydraNorth/pull/1191/files#diff-bc9c26a8a1c0080e54d4d1aae4cc537bR52


Peter Binkley, Ph.D., MLIS / Digital Initiatives Technology Librarian /
peter.binkley@ualberta.ca
2-10K Cameron Library / University of Alberta / Edmonton, Alberta / Canada
T6G 2J8
phone 780-492-3743 / fax 780-492-9243

On 30 June 2016 at 11:13, Weiwei Shi <notifications@github.com> wrote:

> In app/helpers/generic_file_helper.rb
> <https://github.com/ualbertalib/HydraNorth/pull/1191#discussion_r69172152>
> :
>
> > @@ -39,7 +43,22 @@ def download_path(*args)
> >        gf = GenericFile.find(item.id)
> >        return gf.doi_url if gf.doi_url.present?
> >      end
> > -    return sufia.download_path(*args)
> > +    if args[1].nil?
>
> @pbinkley <https://github.com/pbinkley> what is args[1]?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/pull/1191/files/f4aedb48e8deb77e9fdf71db6fe43d90ce0497bb#r69172152>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAQpsILBaGZsm3Sfw86HjF_X80ZuvGCBks5qQ_lLgaJpZM4Iyegz>
> .
>